coalib/core/Bear.py

```diff
@@ -179,23 +196,24 @@ def maintainers_emails(cls):
                 if cls.MAINTAINERS_EMAILS == set() else
                 cls.MAINTAINERS_EMAILS)
 
-    @enforce_signature
-    def __init__(self, section: Section, file_dict: dict):
+    @staticmethod
+    def setup_dependencies():
+        """
+        This is a user defined function that can download and set up
+        dependencies (via download_cached_file or arbitrary other means) in an
+        OS independent way.
         """
-        Constructs a new bear.
 
-        :param section:
-            The section object where bear settings are contained.
-        :param file_dict:
-            The file-dictionary containing a mapping of filenames to the
-            according file contents.
-        :raises RuntimeError:
-            Raised when bear requirements are not fulfilled.
+    def __attrs_post_init__(self):
+        """
+        This function is called after all attributes are initialized.
+        It checks for pre-requisites.
         """
-        self.section = section
-        self.file_dict = file_dict
 
-        self._dependency_results = defaultdict(list)
+        if not isinstance(self.section, Section):
```

Comment:
added manual isinstance check as enforce_signature is not allowed for Attributes and tests were there to assert TypeErrors. This can be improved by including enforce_signature support for Attributes in coala-utils. @Makman2  is it okay for now?